WMF Framework/ExploreCardCollectionViewCell.swift

```diff
@@ -226,24 +225,30 @@ public class ExploreCardCollectionViewCell: CollectionViewCell, Themeable {
                 cardBackgroundView.frame = cardContentViewFrame.insetBy(dx: -singlePixelDimension, dy: -singlePixelDimension)
             }
             origin.y += cardContentViewFrame.layoutHeight(with: 20)
-        } else if isCollapsed {
+        }
+
+        if isCollapsed, !undoLabel.isHidden, !undoButton.isHidden {
+            let undoInsets: (x: CGFloat, y: CGFloat) = (x: 15, y: 16)
+            labelOrigin.x += undoInsets.x
+            labelOrigin.y += undoInsets.y
+
+            let undoLabelMaxWidth = widthMinusMargins - (widthMinusMargins * 0.25)
+            let undoLabelMinWidth = widthMinusMargins * 0.5
+            let undoLabelX = isRTL ? widthMinusMargins - undoLabelMaxWidth : labelOrigin.x
+            let undoLabelFrameHeight = undoLabel.wmf_preferredHeight(at: CGPoint(x: undoLabelX, y: labelOrigin.y), maximumWidth: undoLabelMaxWidth, minimumWidth: undoLabelMinWidth, horizontalAlignment: labelHorizontalAlignment, spacing: 0, apply: apply)
+
+            let undoButtonMaxWidth = widthMinusMargins * 0.25
+            let undoButtonX = isRTL ? labelOrigin.x : widthMinusMargins - undoButtonMaxWidth
+            let undoButtonFrameHeight = undoButton.wmf_preferredHeight(at: CGPoint(x: undoButtonX, y: labelOrigin.y), maximumWidth: undoButtonMaxWidth, horizontalAlignment: buttonHorizontalAlignment, spacing: 0, apply: apply)
+
+            let undoHeight = max(undoLabelFrameHeight, undoButtonFrameHeight)
+            let cardBackgroundViewHeight = ceil(origin.y) + undoHeight + undoInsets.y
```

Comment:
What's the reason for `origin.y` here? Conceptually it seems like `cardBackgroundViewHeight` should be `undoHeight + 2*undoInsets.y`